pipeline_dp/budget_accounting.py

```diff
@@ -104,6 +109,64 @@ def request_budget(
     def compute_budgets(self):
         pass
 
+    def scope(self, weight: float):
+        """
+        Defines a scope for DP operations that should consume no more than "weight" proportion of the budget
+        of the parent scope. The accountant will automatically scale the budgets of all sub-operations accordingly.
+
+        Example usage:
+          with accountant.scope(weight = 0.5):
+             ... some code that consumes DP budget ...
+
+        :param weight: budget weight of all operations made within this scope as compared to.
```

Comment:
According to [style guide](https://google.github.io/styleguide/pyguide.html#244-decision):

Args:
  weight:

Returns:
 ...